lib/datasets/geolonia.rb

```diff
@@ -0,0 +1,55 @@
+require 'csv'
+
+require_relative 'dataset'
+
+module Datasets
+  class Geolonia < Dataset
+    Record = Struct.new(:prefecture_code,
+                        :prefecture_name,
+                        :prefecture_kana,
+                        :prefecture_romaji,
+                        :municipality_code,
+                        :municipality_name,
+                        :municipality_kana,
+                        :municipality_romaji,
+                        :street_name,
+                        :street_kana,
+                        :street_romaji,
+                        :alias,
+                        :latitude,
+                        :longitude)
+
+    def initialize
+      super
+      @metadata.id = 'geolonia'
+      @metadata.name = 'Geolonia'
+      @metadata.url = 'https://github.com/geolonia/japanese-addresses'
+      @metadata.licenses = ["CC-BY-4.0"]
+      @metadata.description = <<~DESCRIPTION
```

Comment:
How about using the part of README?

```diff
diff --git a/lib/datasets/geolonia.rb b/lib/datasets/geolonia.rb
index 52fe73b..0b9f25e 100644
--- a/lib/datasets/geolonia.rb
+++ b/lib/datasets/geolonia.rb
@@ -25,8 +25,9 @@ module Datasets
       @metadata.name = 'Geolonia'
       @metadata.url = 'https://github.com/geolonia/japanese-addresses'
       @metadata.licenses = ["CC-BY-4.0"]
-      @metadata.description = <<~DESCRIPTION
-    DESCRIPTION
+      @metadata.description = lambda do
+        fetch_readme
+      end
     end
 
     def each
@@ -42,14 +43,25 @@ module Datasets
     end
 
     private
+    def download_base_url
+      "https://raw.githubusercontent.com/geolonia/japanese-addresses/master"
+    end
 
     def open_data
       data_path = cache_dir_path + 'latest.csv'
-      data_url = 'https://raw.githubusercontent.com/geolonia/japanese-addresses/develop/data/latest.csv'
+      data_url = "#{download_base_url}/data/latest.csv"
       download(data_path, data_url)
       CSV.open(data_path) do |csv|
         yield(csv)
       end
     end
+
+    def fetch_readme
+      readme_base_name = "README.md"
+      readme_path = cache_dir_path + readme_base_name
+      readme_url = "#{download_base_url}/#{readme_base_name}"
+      download(readme_path, readme_url)
+      readme_path.read.split(/^## API/, 2)[0].strip
+    end
   end
 end
diff --git a/test/test-geolonia.rb b/test/test-geolonia.rb
index 8346628..7c03d1b 100644
--- a/test/test-geolonia.rb
+++ b/test/test-geolonia.rb
@@ -46,4 +46,18 @@ class GeoloniaTest < Test::Unit::TestCase
                    records[-1].to_h,
                  ])
   end
+
+  sub_test_case("#metadata") do
+    test("#description") do
+      description = @dataset.metadata.description
+      assert_equal([
+                     "# Geolonia 住所データ",
+                     "## 住所データ仕様",
+                     "### ファイルフォーマット",
+                     "### 列",
+                   ],
+                   description.scan(/^#.*$/),
+                   description)
+    end
+  end
 end
```

Reply:
Thank you!

Reply:
I've updated the patch for the latest revision:

```diff
diff --git a/lib/datasets/geolonia.rb b/lib/datasets/geolonia.rb
index c45eb5c..0b9f25e 100644
--- a/lib/datasets/geolonia.rb
+++ b/lib/datasets/geolonia.rb
@@ -25,8 +25,9 @@ module Datasets
       @metadata.name = 'Geolonia'
       @metadata.url = 'https://github.com/geolonia/japanese-addresses'
       @metadata.licenses = ["CC-BY-4.0"]
-      @metadata.description = <<~DESCRIPTION
-    DESCRIPTION
+      @metadata.description = lambda do
+        fetch_readme
+      end
     end
 
     def each
@@ -42,14 +43,25 @@ module Datasets
     end
 
     private
+    def download_base_url
+      "https://raw.githubusercontent.com/geolonia/japanese-addresses/master"
+    end
 
     def open_data
       data_path = cache_dir_path + 'latest.csv'
-      data_url = 'https://raw.githubusercontent.com/geolonia/japanese-addresses/master/data/latest.csv'
+      data_url = "#{download_base_url}/data/latest.csv"
       download(data_path, data_url)
       CSV.open(data_path) do |csv|
         yield(csv)
       end
     end
+
+    def fetch_readme
+      readme_base_name = "README.md"
+      readme_path = cache_dir_path + readme_base_name
+      readme_url = "#{download_base_url}/#{readme_base_name}"
+      download(readme_path, readme_url)
+      readme_path.read.split(/^## API/, 2)[0].strip
+    end
   end
 end
diff --git a/test/test-geolonia.rb b/test/test-geolonia.rb
index 8346628..7c03d1b 100644
--- a/test/test-geolonia.rb
+++ b/test/test-geolonia.rb
@@ -46,4 +46,18 @@ class GeoloniaTest < Test::Unit::TestCase
                    records[-1].to_h,
                  ])
   end
+
+  sub_test_case("#metadata") do
+    test("#description") do
+      description = @dataset.metadata.description
+      assert_equal([
+                     "# Geolonia 住所データ",
+                     "## 住所データ仕様",
+                     "### ファイルフォーマット",
+                     "### 列",
+                   ],
+                   description.scan(/^#.*$/),
+                   description)
+    end
+  end
 end
```

Reply:
Can you please confirm this?
Thank you for your time.

```
$ git diff
warning: LF will be replaced by CRLF in lib/datasets/geolonia.rb.
The file will have its original line endings in your working directory
warning: LF will be replaced by CRLF in test/test-geolonia.rb.
The file will have its original line endings in your working directory
diff --git a/lib/datasets/geolonia.rb b/lib/datasets/geolonia.rb
index c45eb5c..0b9f25e 100644
--- a/lib/datasets/geolonia.rb
+++ b/lib/datasets/geolonia.rb
@@ -25,8 +25,9 @@ module Datasets
       @metadata.name = 'Geolonia'
       @metadata.url = 'https://github.com/geolonia/japanese-addresses'
       @metadata.licenses = ["CC-BY-4.0"]
-      @metadata.description = <<~DESCRIPTION
-    DESCRIPTION
+      @metadata.description = lambda do
+        fetch_readme
+      end
     end

     def each
@@ -42,14 +43,25 @@ module Datasets
     end

     private
+    def download_base_url
+      "https://raw.githubusercontent.com/geolonia/japanese-addresses/master"
+    end

     def open_data
       data_path = cache_dir_path + 'latest.csv'
-      data_url = 'https://raw.githubusercontent.com/geolonia/japanese-addresses/master/data/latest.csv'
+      data_url = "#{download_base_url}/data/latest.csv"
       download(data_path, data_url)
       CSV.open(data_path) do |csv|
         yield(csv)
       end
     end
+
+    def fetch_readme
+      readme_base_name = "README.md"
+      readme_path = cache_dir_path + readme_base_name
+      readme_url = "#{download_base_url}/#{readme_base_name}"
+      download(readme_path, readme_url)
+      readme_path.read.split(/^## API/, 2)[0].strip
+    end
   end
 end
diff --git a/test/test-geolonia.rb b/test/test-geolonia.rb
index 8346628..07b7db3 100644
--- a/test/test-geolonia.rb
+++ b/test/test-geolonia.rb
@@ -3,6 +3,20 @@ class GeoloniaTest < Test::Unit::TestCase
     @dataset = Datasets::Geolonia.new
   end

+  sub_test_case("#metadata") do
+    test("#description") do
+      description = @dataset.metadata.description
+      assert_equal([
+                     "# Geolonia 住所データ",
+                     "## 住所データ仕様",
+                     "### ファイルフォーマット",
+                     "### 列",
+                   ],
+                   description.scan(/^#.*$/),
+                   description)
+    end
+  end
+
   test('#each') do
     records = @dataset.each.to_a
     assert_equal([
```